tests/suite/snippets_filter_test.go

```diff
@@ -0,0 +1,345 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	ngfAPI "github.com/nginxinc/nginx-gateway-fabric/apis/v1alpha1"
+	"github.com/nginxinc/nginx-gateway-fabric/internal/mode/static/state/conditions"
+	"github.com/nginxinc/nginx-gateway-fabric/tests/framework"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	core "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	v1 "sigs.k8s.io/gateway-api/apis/v1"
+)
+
+var _ = Describe("SnippetsFilter", Ordered, Label("functional", "snippets-filter"), func() {
+	var (
+		files = []string{
+			"snippets-filter/cafe.yaml",
+			"snippets-filter/gateway.yaml",
+			"snippets-filter/grpc-backend.yaml",
+			"snippets-filter/cafe-routes.yaml",
+			"snippets-filter/grpc-route.yaml",
+		}
+
+		namespace = "snippets-filter"
+	)
+
+	BeforeAll(func() {
+		ns := &core.Namespace{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: namespace,
+			},
+		}
+
+		Expect(resourceManager.Apply([]client.Object{ns})).To(Succeed())
+		Expect(resourceManager.ApplyFromFiles(files, namespace)).To(Succeed())
+		Expect(resourceManager.WaitForAppsToBeReady(namespace)).To(Succeed())
+
+	})
+
+	AfterAll(func() {
+		Expect(resourceManager.DeleteNamespace(namespace)).To(Succeed())
+	})
+
+	When("SnippetsFilters are applied to the resources", func() {
+		var (
+			snippetsFilter = []string{
+				"snippets-filter/valid-sf.yaml",
+			}
+			baseURL string
+		)
+
+		BeforeAll(func() {
+			Expect(resourceManager.ApplyFromFiles(snippetsFilter, namespace)).To(Succeed())
+
+			port := 80
+			if portFwdPort != 0 {
+				port = portFwdPort
+			}
+			baseURL = fmt.Sprintf("http://cafe.example.com:%d%s", port, "/coffee")
+		})
+
+		AfterAll(func() {
+			Expect(resourceManager.DeleteFromFiles(snippetsFilter, namespace)).To(Succeed())
+		})
+
+		Specify("snippetsFilters are accepted", func() {
+			snippetsFilterNames := []string{
+				"all-contexts",
+				"grpc-all-contexts",
+			}
+
+			for _, name := range snippetsFilterNames {
+				nsname := types.NamespacedName{Name: name, Namespace: namespace}
+
+				err := waitForSnippetsFilterToBeAccepted(nsname)
+				Expect(err).ToNot(HaveOccurred(), fmt.Sprintf("%s was not accepted", name))
+			}
+		})
+
+		Context("verify working traffic", func() {
+			It("should return a 200 response for HTTPRoute", func() {
+				Eventually(
+					func() error {
+						return expectRequestToSucceed(baseURL, address, "URI: /coffee")
+					}).
+					WithTimeout(timeoutConfig.CreateTimeout).
+					WithPolling(5 * time.Second).
```

Comment:
```suggestion
					WithPolling(500 * time.Millisecond).
```
We don't need to wait 5 seconds between each request, that would take too long.

Reply:
The routes take a lot of time for the status updated and the polling too quickly doesn't really help. so I spaced them apart!

Reply:
Well `WaitForAppsToBeReady` should already be checking for the route to be ready, so I wouldn't expect this test to even start until the HTTPRoute is ready.

Reply:
This test also just sends traffic, it doesn't check for status.

Reply:
Yeah I misunderstood the comment !